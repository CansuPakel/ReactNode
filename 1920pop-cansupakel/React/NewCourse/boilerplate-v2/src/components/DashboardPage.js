import React from 'react';

const DashboardPage = (props) =>
{
    return(
        <div>
        </div>
    )
}

export default DashboardPage;