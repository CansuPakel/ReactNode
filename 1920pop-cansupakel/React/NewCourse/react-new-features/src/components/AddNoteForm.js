import React, { useState, useContext } from "react";
import NoteContext from '../context/notes-context';

const AddNoteForm = () => {
    const {dispatch} = useContext(NoteContext);
  const [title, setTitle] = useState("");
  const [body, setBody] = useState("");

  const addNote = e => {
    e.preventDefault();
    dispatch({
      type: "ADD_NOTE",
      title,
      body
    });
    setTitle(""); //remove wat in inputstaat
    setBody("");
  };

  return (
    <form onSubmit={addNote}>
      <input value={title} onChange={e => setTitle(e.target.value)} />
      <textarea value={body} onChange={e => setBody(e.target.value)}></textarea>
      <button>add note</button>
    </form>
  );
};

export default AddNoteForm;
