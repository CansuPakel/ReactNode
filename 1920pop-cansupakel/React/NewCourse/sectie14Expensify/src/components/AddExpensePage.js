import React from 'react';
import ExpenseForm from './ExpenseForm'
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expenses'

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense)
        this.props.history.push('/'); //gaat naar dashboard na een het toevoegen
    };

    render(){
        return (
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Voeg een expense toe</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm
                onSubmit={this.onSubmit}
                />
            </div>
        </div>
        )
    }
}

//dit is gemakkelijker om te testen
const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});


export default connect(undefined, mapDispatchToProps)(AddExpensePage);