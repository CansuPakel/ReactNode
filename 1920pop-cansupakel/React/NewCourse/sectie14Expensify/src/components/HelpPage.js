import React from 'react';
const HelpPage = () =>
{
    return(
        <div>
            this is from my HelpPage
        </div>
    )
}

export default HelpPage;