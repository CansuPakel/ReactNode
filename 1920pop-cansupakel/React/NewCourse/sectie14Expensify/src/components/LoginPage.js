import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage = ({startLogin}) => {
        return(
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Login page React-POP</h1>
                    <p>expenses</p>
                    <button  className="button" onClick={startLogin}>Login met Google</button>
                </div>
            </div>
        )
}

const mapDispatchToProps = (dispatch) =>({
    startLogin: () => dispatch(startLogin())
})
export default connect(undefined,mapDispatchToProps)(LoginPage);