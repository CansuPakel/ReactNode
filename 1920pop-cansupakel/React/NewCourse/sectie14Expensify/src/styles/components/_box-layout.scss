.box-layout {
    background:url('/images/bg.jpg');
    background-size:cover; //volledige container
    display: flex;
    justify-content: center;
    align-items: center;
    height:100vh;
    width:100vw; //viewport
}

.box-layout__box {
    background: fade-out(white , .15);
    border-radius: 4px;
    text-align: center;
    width:30rem;
    padding: $s-size $m-size;
}

.box-layout__title {
    margin:0 0 $m-size 0; //bottom
    line-height: 1;
}