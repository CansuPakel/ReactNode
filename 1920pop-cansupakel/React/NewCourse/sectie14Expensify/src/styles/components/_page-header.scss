.page-header{
    background: #f7f7f7;
    margin-bottom:$l-size;
    padding:$l-size 0; //Bttom
}


.page-header__title{
    margin:0;
    font-weight: 300;
    span {
        font-weight: bold;
    }
}

.page-header__actions{
    margin-top:$m-size;
}