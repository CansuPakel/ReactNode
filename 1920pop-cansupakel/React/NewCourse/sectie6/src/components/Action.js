 import React from 'react';
 //geen this gebruiken en het renderd vanzelf dus geen render()
 const Action = (props) => {
    return (
      <div>
        <button
         disabled={!props.hasOptions} 
         onClick={props.handleAction}>
         What should I do?
         </button>
      </div>
    )
  }
  
  /**class Action extends React.Component {
    render(){
      return (
        <div>
          <button
           disabled={!this.props.hasOptions} 
           onClick={this.props.handleAction}>
           What should I do?
           </button>
        </div>
      )
    }
  }*/

  export default Action;