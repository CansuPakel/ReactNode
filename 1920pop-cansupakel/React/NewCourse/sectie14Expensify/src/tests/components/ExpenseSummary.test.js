import React from 'react';
import {shallow} from 'enzyme';
import {  ExpenseSummary }  from '../../components/ExpenseSummary';

test('should render with 1 expense',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={230}/>)
    expect(wrapper).toMatchSnapshot();
})

test('should render with more expense',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={2435665}/>)
    expect(wrapper).toMatchSnapshot();
})