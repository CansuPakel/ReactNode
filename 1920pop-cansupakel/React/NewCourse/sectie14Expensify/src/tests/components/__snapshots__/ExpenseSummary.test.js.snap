// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
      : 
      <span>
        $2.30
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`should render with more expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
      : 
      <span>
        $24,356.65
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add expense
      </[object Object]>
    </div>
  </div>
</div>
`;
